src/client/Microsoft.Identity.Client/Internal/ClientCredential/SignedAssertionDelegateClientCredential.cs

```diff
@@ -42,7 +42,10 @@ public async Task AddConfidentialClientParametersAsync(
             {
                 // If no "AssertionRequestOptions" delegate is supplied
                 string signedAssertion = await _signedAssertionDelegate(cancellationToken).ConfigureAwait(false);
-                oAuth2Client.AddBodyParameter(OAuth2Parameter.ClientAssertionType, OAuth2AssertionType.JwtBearer);
+
+                oAuth2Client.AddBodyParameter(OAuth2Parameter.ClientAssertionType,
+                                                                requestParameters.AppConfig.ClientId == Constants.FmiUrnClientId ?
+                                                                OAuth2AssertionType.FmiBearer : OAuth2AssertionType.JwtBearer);
                 oAuth2Client.AddBodyParameter(OAuth2Parameter.ClientAssertion, signedAssertion);
             }
             else
```

Comment:
You need to update at line 77 :( as well. It's a valid code path. It's for one of the overloads of WithClientAssertion

Reply:
Oh yes I see now, I though that path was from an obsolete api, I misread the code

Reply:
Just fixed it, no worries.